src/App.js

```diff
@@ -6,32 +6,39 @@ import { useEffect, useState } from "react";
 import axios from "axios";
 
 const App = () => {
-  const [inputLanguage, setInputLanguage] = useState("English");
-  const [outputLanguage, setOutputLanguage] = useState("Japanese");
+  const [inputLanguage, setInputLanguage] = useState("English (en)");
+  const [outputLanguage, setOutputLanguage] = useState("Japanese (ja)");
   const [showModal, setShowModal] = useState(null);
   const [languages, setLanguages] = useState(null);
   const [textToTranslate, setTextToTranslate] = useState("");
   const [translatedText, setTranslatedText] = useState("");
 
-  const getLanguages = async() => {
+  const extractLanguageCode = (languageString) => {
+    const match = languageString.match(/\((\w+)\)/); // This regex matches the content within parentheses.
+    if (match && match[1]) {
+      return match[1]; // Returns the matched language code (in this case, "en").
+    }
+    return null; // Return null if no match is found.
+  }
+
+  const getLanguages = async () => {
     const response = await axios.get("http://localhost:8000/languages");
     setLanguages(response.data);
   }
 
   const translate = async () => {
     const data = {
-      textToTranslate, outputLanguage, inputLanguage
+      textToTranslate,
+      outputLanguage: extractLanguageCode(outputLanguage),
+      inputLanguage: extractLanguageCode(inputLanguage)
     }
```

Comment:
**This was the main problem your app was having:** You already mapped each language response as `language.name (language.code)` **BUT** your API endpoint requires the **language codes only** 

i.e `translator.translateText(textToTranslate, inputLanguage, outputLanguage)` requires `inputLanguage` and `outputLanguage` to be `en` or any other code, BUT because you already formatted it to return `English (en)`, this was what was sending to `translator.translateText`, which causes the error.

Reply:
I got it. 
As you said, I didn’t know how to extract the language code using a function. 
Many thanks!